crates/fsrt/src/forge_project.rs

```diff
@@ -0,0 +1,197 @@
+                        decorators: true,
+                        ..Default::default()
+                    }),
+                    target,
+                    None,
+                    &mut recovered_errors,
+                )
+                .unwrap();
+                debug!("finished parsing");
+                let mut hygeine = resolver(Mark::new(), Mark::new(), true);
+                let module = module.fold_with(&mut hygeine);
+                ctx.load_module(p, module);
+                ctx
+            })
+        });
+        let keys = ctx.module_ids().collect::<Vec<_>>();
+        debug!(?keys);
+        let env = run_resolver(ctx.modules(), ctx.file_resolver(), secret_packages);
+        ForgeProject {
+            sm,
+            ctx,
+            env,
+            funcs: vec![],
+        }
+    }
+
+    fn get_paths(&self) -> HashSet<PathBuf>;
+
+    #[allow(dead_code)]
+    fn get_secret_packages(&self) -> Vec<PackageData>;
+
+    fn get_manifest(&self) -> ForgeManifest<'_>;
+}
+pub(crate) struct ForgeProject<'a> {
+    #[allow(dead_code)]
+    pub sm: Arc<SourceMap>,
+    pub ctx: AppCtx,
+    pub env: Environment,
+    pub funcs: Vec<ResolvedEntryPoint<'a>>,
+}
+
+impl<'a> ForgeProject<'a> {
+    #[inline]
+    pub fn add_funcs<I: IntoIterator<Item = Entrypoint<'a, Resolved>>>(&mut self, iter: I) {
+        self.funcs.extend(iter.into_iter().filter_map(|entrypoint| {
+            let (func_name, path) = entrypoint.function.into_func_path();
+            let module = self.ctx.modid_from_path(&path)?;
+            let def_id = self.env.module_export(module, func_name)?;
+            Some(ResolvedEntryPoint {
+                func_name,
+                path,
+                module,
+                def_id,
+                invokable: entrypoint.invokable,
+                webtrigger: entrypoint.web_trigger,
+                admin: entrypoint.admin,
+            })
+        }));
+    }
+}
+
+#[allow(dead_code)]
+impl MockForgeProject<'_> {
+    pub fn files_from_string(string: String) -> Self {
+        let forge_manifest = ForgeManifest::create_manifest_with_func_mod(FunctionMod {
+            key: "main",
+            handler: "index.run",
+            providers: None,
+        });
+
+        let mut mock_forge_project = MockForgeProject {
+            files_name_to_source: HashMap::new(),
+            test_manifest: forge_manifest,
+        };
+
+        let different_files = string
+            .split("//")
+            .map(|f| f.replace("//", "").trim().to_string())
+            .filter(|file| !file.is_empty());
+        for file in different_files {
+            let (file_name, file_source) = file.split_once('\n').unwrap();
+            mock_forge_project
+                .files_name_to_source
+                .insert(PathBuf::from(file_name.trim()), file_source.to_string());
+        }
+
+        mock_forge_project
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct ForgeProjectFromDir {
+    #[allow(dead_code)]
```

Comment:
You can just delete this `dir` property if it isn't being used anywhere.

Reply:
This actually was being used. I have no idea why I did this....